lib/modules/datasource/docker/index.ts

```diff
@@ -201,6 +208,9 @@ export async function getAuthHeaders(
     if (err.statusCode >= 500 && err.statusCode < 600) {
       throw new ExternalHostError(err);
     }
+    if (err.message === PAGE_NOT_FOUND_ERROR) {
+      return {};
+    }
```

Comment:
```suggestion
    if (err.message === PAGE_NOT_FOUND_ERROR) {
      throw err;
    }
```

we should not try to use empty headers if we've a 404.

needs a change here
https://github.com/renovatebot/renovate/blob/5774fb3f2a09e42f3efbac8169cdbed78502220f/lib/modules/datasource/docker/index.ts#L770

```ts
if ((err.statusCode === 404 || err.message === PAGE_NOT_FOUND_ERROR) && !dockerRepository.includes('/'))
```